enclave-tls/src/tls_wrappers/api/tls_wrapper_verify_certificate_extension.c

```diff
@@ -16,22 +16,24 @@ tls_wrapper_err_t tls_wrapper_verify_certificate_extension(tls_wrapper_ctx_t *tl
 {
 	ETLS_DEBUG("tls_wrapper_verify_certificate_extension() called with evidence type: '%s'\n", evidence->type);
 
-	if (!tls_ctx || !tls_ctx->etls_handle || !tls_ctx->etls_handle->verifier || !tls_ctx->etls_handle->verifier->opts ||
-			!tls_ctx->etls_handle->verifier->opts->verify_evidence)
+	if (!tls_ctx || !tls_ctx->etls_handle)
 		return -TLS_WRAPPER_ERR_INVALID;
 
-	if (strcmp(tls_ctx->etls_handle->verifier->opts->type, evidence->type)) {
-		ETLS_WARN("type doesn't match between verifier '%s' and evidence '%s'\n", tls_ctx->etls_handle->verifier->opts->type, evidence->type);
+	if (tls_ctx->etls_handle->flags & ENCLAVE_TLS_CONF_VERPARA_AUTO) {
 		enclave_tls_err_t tlserr = etls_verifier_select(tls_ctx->etls_handle, evidence->type, tls_ctx->etls_handle->config.cert_algo);
 		if (tlserr != ENCLAVE_TLS_ERR_NONE) {
 			ETLS_ERR("the verifier selecting err %#x during verifying cert extension\n", tlserr);
 			return -TLS_WRAPPER_ERR_INVALID;
 		}
 	}
 
-	enclave_quote_err_t err =
-		tls_ctx->etls_handle->verifier->opts->verify_evidence(tls_ctx->etls_handle->verifier, evidence,
-				hash, hash_len);
+	if (!tls_ctx->etls_handle->verifier || !tls_ctx->etls_handle->verifier->opts || !tls_ctx->etls_handle->verifier->opts->verify_evidence)
+		return -TLS_WRAPPER_ERR_INVALID;
+
+	if (strcmp(tls_ctx->etls_handle->verifier->opts->type, evidence->type))
+		ETLS_WARN("type doesn't match between verifier '%s' and evidence '%s'\n", tls_ctx->etls_handle->verifier->opts->type, evidence->type);
+
+	enclave_quote_err_t err = tls_ctx->etls_handle->verifier->opts->verify_evidence(tls_ctx->etls_handle->verifier, evidence, hash, hash_len);
```

Comment:
Change to:

```C
if (strcmp(tls_ctx->etls_handle->verifier->opts->type, evidence->type) && !(tls_ctx->etls_handle->flags & ENCLAVE_TLS_CONF_VERIFIER_ENFORCED)) {
	enclave_tls_err_t tlserr = etls_verifier_select(tls_ctx-&gt;etls_handle, evidence-&gt;type, tls_ctx-&gt;etls_handle-&gt;config.cert_algo);
...
}

err = tls_ctx->etls_handle->verifier->opts->verify_evidence(tls_ctx->etls_handle->verifier, evidence, hash, hash_len);
```

This makes the codes simpler.

Reply:
please refer to my previous comments.

Reply:
Essentially, our modifications in this portion are same:

Yours:
```C
if (flags & AUTO)
  select();

verify();
```

Mine:
```C
if (type_unmatched && !(flags & ENFORCED))
  select();

verify();
```

So the final decision is related to the previous portion showing the thoughts and ideas based on our insights between us.